### DISCUSSION TITLE
Binary files lacking the "filename" attribute are mangled if passed as bytes and unprocessable if passed as UploadFile

### DISCUSSION THREAD

**Discussion by alexhenrie** [fastapi/fastapi#9629] [Questions]
### First Check

- [X] I added a very descriptive title here.
- [X] I used the GitHub search to find a similar question and didn't find it.
- [X] I searched the FastAPI documentation, with the integrated search.
- [X] I already searched in Google "How to X in FastAPI" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to FastAPI but to [Pydantic](https://github.com/pydantic/pydantic).
- [X] I already checked if it is not related to FastAPI but to [Swagger UI](https://github.com/swagger-api/swagger-ui).
- [X] I already checked if it is not related to FastAPI but to [ReDoc](https://github.com/Redocly/redoc).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
from fastapi import FastAPI, File, UploadFile
from fastapi.testclient import TestClient
from io import BytesIO

app = FastAPI()
client = TestClient(app)

@app.post('/bytes')
async def set_ss_creds(f: bytes = File()):
    print('    Length', len(f), 'Contents', f)

@app.post('/upload-file')
async def set_ss_creds(f: UploadFile = File()):
    contents = f.file.read()
    print('    Length', len(contents), 'Contents', contents)

print('As bytes with file name:')
client.post('/bytes', files={'f': BytesIO(b'\xff')})

print('As bytes without file name:')
client.post('/bytes', files={'f': (None, BytesIO(b'\xff'), 'application/octet-stream')})

print('As UploadFile with file name:')
client.post('/upload-file', files={'f': BytesIO(b'\xff')})

print('As UploadFile without file name:')
print('   ', client.post('/upload-file', files={'f': (None, BytesIO(b'\xff'), 'application/octet-stream')}))
```

### Description

Expected results:

```
As bytes with file name:
    Length 1 Contents b'\xff'
As bytes without file name:
    Length 1 Contents b'\xff'
As UploadFile with file name:
    Length 1 Contents b'\xff'
As UploadFile without file name:
    Length 1 Contents b'\xff'
    <Response [200 OK]>
```

Actual results:

```
As bytes with file name:
    Length 1 Contents b'\xff'
As bytes without file name:
    Length 2 Contents b'\xc3\xbf'
As UploadFile with file name:
    Length 1 Contents b'\xff'
As UploadFile without file name:
    <Response [422 Unprocessable Entity]>
```

To give you a better idea of what's going on... When the file name is specified, the request includes the filename attribute:

```
Content-Disposition: form-data; name="f"; filename="upload"
Content-Type: application/octet-stream
```

When it is not set, the request lacks the filename attribute:

```
Content-Disposition: form-data; name="f"
Content-Type: application/octet-stream
```

As you can see, if the file is sent with the "name" attribute but lacks "filename", FastAPI encodes its content as UTF-8 before passing it as bytes ([U+00FF](https://www.compart.com/en/unicode/U+00FF) => `C3 BF`), or refuses to process it at all if passing it as an UploadFile.

In my opinion, FastAPI should always pass the file contents without modification and without error. However, it would also be acceptable if it refused to ever process files that lack a filename attribute.

### Operating System

Linux

### Operating System Details

Arch Linux

### FastAPI Version

0.95.2

### Python Version

Python 3.11.3

**Comment by ijessen-mitll** [comment]
Holy smokes THANK YOU for capturing the solution to this!!  I was questioning my sanity in debugging it.  I would agree wholeheartedly that bytes *should not* be mangled - ever.
